frame/support/test/tests/pallet_compatibility_instance.rs

```diff
@@ -0,0 +1,311 @@
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use sp_runtime::traits::Block as _;
+
+mod pallet_old {
+	use frame_support::{
+		decl_storage, decl_error, decl_event, decl_module, weights::Weight, traits::Get, Parameter
+	};
+	use frame_system::ensure_root;
+
+	pub trait Trait<I: Instance = DefaultInstance>: frame_system::Trait {
+		type SomeConst: Get<Self::Balance>;
+		type Balance: Parameter + codec::HasCompact + From<u32> + Into<Weight> + Default;
+		type Event: From<Event<Self, I>> + Into<<Self as frame_system::Trait>::Event>;
+	}
+
+	decl_storage! {
+		trait Store for Module<T: Trait<I>, I: Instance = DefaultInstance> as Example {
+			/// Some documentation
+			Dummy get(fn dummy) config(): Option<T::Balance>;
+			Bar get(fn bar) config(): map hasher(blake2_128_concat) T::AccountId => T::Balance;
+			Foo get(fn foo) config(): T::Balance = 3.into();
+			Double get(fn double): double_map hasher(blake2_128_concat) u32, hasher(twox_64_concat) u64 => u16;
+		}
+	}
+
+	decl_event!(
+		pub enum Event<T, I = DefaultInstance> where Balance = <T as Trait<I>>::Balance {
+			/// Dummy event, just here so there's a generic type that's used.
+			Dummy(Balance),
+		}
+	);
+
+	decl_module! {
+		pub struct Module<T: Trait<I>, I: Instance = DefaultInstance> for enum Call where origin: T::Origin {
+			type Error = Error<T, I>;
+			fn deposit_event() = default;
+			const SomeConst: T::Balance = T::SomeConst::get();
+
+			#[weight = <T::Balance as Into<Weight>>::into(new_value.clone())]
+			fn set_dummy(origin, #[compact] new_value: T::Balance) {
+				ensure_root(origin)?;
+
+				<Dummy<T, I>>::put(&new_value);
+				Self::deposit_event(RawEvent::Dummy(new_value));
+			}
+
+			fn on_initialize(_n: T::BlockNumber) -> Weight {
+				<Dummy<T, I>>::put(T::Balance::from(10));
+				10
+			}
+
+			fn on_finalize(_n: T::BlockNumber) {
+				<Dummy<T, I>>::put(T::Balance::from(11));
+			}
+		}
+	}
+
+	decl_error! {
+		pub enum Error for Module<T: Trait<I>, I: Instance> {
+			/// Some wrong behavior
+			Wrong,
+		}
+	}
+}
+
+#[frame_support::pallet(Example)]
+pub mod pallet {
+	use frame_support::pallet_prelude::*;
+	use frame_system::pallet_prelude::*;
+	use frame_system::ensure_root;
+
+	#[pallet::trait_]
+	pub trait Trait<I: Instance = DefaultInstance>: frame_system::Trait {
+		type Balance: Parameter + codec::HasCompact + From<u32> + Into<Weight> + Default
+			+ MaybeSerializeDeserialize;
+		#[pallet::const_]
+		type SomeConst: Get<Self::Balance>;
+		type Event: From<Event<Self, I>> + IsType<<Self as frame_system::Trait>::Event>;
+	}
+
+	#[pallet::module]
```

Comment:
@kianenigma please give a better name here :P

Reply:
as suggested in https://github.com/paritytech/substrate/issues/7194 we can rename to pallet::pallet and have the struct named `struct Pallet<.....` but then we need construct_runtime able to use `Module` or `Pallet`. 

I implemented it, but didn't update the test yet, does Pallet instead of Module makes more sense?

Reply:
it is now Pallet 👍